File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        s=set()

        for num in nums1:
            if num in nums2:
                s.add(num)
        demo=list(s)
        return demo

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>();
        for(int n2:nums2)
        {
            set.add(n2);
        }
        ArrayList<Integer> list = new ArrayList<>();
        {
            for(int n1:nums1)
            {
                if(set.contains(n1)) 
                {
                    list.add(n1);
                    set.remove(n1);
                }
            }
        }

        int[] res = new int[list.size()];

        for(int i=0;i<list.size();i++)
        {
            res[i] = list.get(i);
        }
        return res;
     }
}